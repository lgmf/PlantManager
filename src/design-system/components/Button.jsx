import React from 'react';
import styled from 'styled-components/native';

import { Colors } from '@design-system/palette';
import { Fonts } from '@design-system/typography';

export const ButtonContainer = styled.TouchableOpacity`
  background-color: ${Colors.green};
  align-items: center;
  justify-content: center;
  border-radius: 16px;
  padding: 20px;
`;

const ButtonText = styled.Text`
  fontFamily: ${Fonts.heading};
  fontSize: 16px;
  color: ${Colors.white};
`;

function Button({
  disabled,
  onPress,
  children,
  testID = 'button',
}) {
  return (
    <ButtonContainer
      testID={testID}
      disabled={disabled}
      activeOpacity={0.7}
      onPress={onPress}
    >
      <ButtonText>
        {children}
      </ButtonText>
    </ButtonContainer>
  );
}

export default Button;
