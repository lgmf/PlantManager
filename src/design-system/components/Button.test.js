import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';

import Button from './Button';

describe('design system / components / button', () => {
  const testID = 'button';

  it('should renders correctly', () => {
    const component = render(
      <Button>
        click me
      </Button>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should call on press', () => {
    const onPress = jest.fn();

    const component = render(
      <Button
        testID={testID}
        onPress={onPress}
      />,
    );

    const button = component.getByTestId(testID);
    fireEvent.press(button);

    expect(onPress).toHaveBeenCalledTimes(1);
  });
});
