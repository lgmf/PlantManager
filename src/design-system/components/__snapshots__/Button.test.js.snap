// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`design system / components / button should renders correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#32B768",
      "borderRadius": 16,
      "justifyContent": "center",
      "opacity": 1,
      "paddingBottom": 20,
      "paddingLeft": 20,
      "paddingRight": 20,
      "paddingTop": 20,
    }
  }
  testID="button"
>
  <Text
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontFamily": "Jost_600SemiBold",
          "fontSize": 16,
        },
      ]
    }
  >
    click me
  </Text>
</View>
`;
